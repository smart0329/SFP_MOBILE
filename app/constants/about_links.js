// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    TERMS_OF_SERVICE: 'https://about.mattermost.com/default-terms/',
    PRIVACY_POLICY: 'https://about.mattermost.com/default-privacy-policy/',
};
