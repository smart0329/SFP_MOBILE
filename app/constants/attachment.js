// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const MAX_ATTACHMENT_FOOTER_LENGTH = 300;
export const ATTACHMENT_ICON_HEIGHT = 48;
export const ATTACHMENT_ICON_WIDTH = 36;
