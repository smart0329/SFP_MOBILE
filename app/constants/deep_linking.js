// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    CHANNEL: 'channel',
    PERMALINK: 'permalink',
    OTHER: 'other',
};
