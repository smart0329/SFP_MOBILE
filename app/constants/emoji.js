// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const ALL_EMOJIS = 'all_emojis';
export const MAX_ALLOWED_REACTIONS = 40;