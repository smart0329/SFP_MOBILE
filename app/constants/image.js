// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const IMAGE_MAX_HEIGHT = 350;
export const IMAGE_MIN_DIMENSION = 50;
