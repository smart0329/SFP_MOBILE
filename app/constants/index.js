// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import DeepLinkTypes from './deep_linking';
import DeviceTypes from './device';
import ListTypes from './list';
import NavigationTypes from './navigation';
import ViewTypes, {UpgradeTypes} from './view';

export {
    DeepLinkTypes,
    DeviceTypes,
    ListTypes,
    NavigationTypes,
    UpgradeTypes,
    ViewTypes,
};
