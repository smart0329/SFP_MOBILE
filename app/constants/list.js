// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

const VISIBILITY_CONFIG_DEFAULTS = {
    itemVisiblePercentThreshold: 100,
    waitForInteraction: false,
};

export default {
    VISIBILITY_CONFIG_DEFAULTS,
    VISIBILITY_SCROLL_DOWN: 'down',
    VISIBILITY_SCROLL_UP: 'up',
};
