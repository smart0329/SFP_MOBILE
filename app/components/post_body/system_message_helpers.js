// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {Posts} from 'mattermost-redux/constants';
import Markdown from 'app/components/markdown';
import {t} from 'app/utils/i18n';

const renderUsername = (value) => {
    return (value[0] === '@') ? value : `@${value}`;
};

const renderMessage = (postBodyProps, styles, intl, localeHolder, values) => {
    const {onPress} = postBodyProps;
    const {messageStyle, textStyles} = styles;

    return (
        <Markdown
            baseTextStyle={messageStyle}
            onPostPress={onPress}
            textStyles={textStyles}
            value={intl.formatMessage(localeHolder, values)}
        />
    );
};

const renderHeaderChangeMessage = (postBodyProps, styles, intl) => {
    const {postProps} = postBodyProps;
    let values;

    if (!postProps.username) {
        return null;
    }

    const username = renderUsername(postProps.username);
    const oldHeader = postProps.old_header;
    const newHeader = postProps.new_header;
    let localeHolder;

    if (postProps.new_header) {
        if (postProps.old_header) {
            localeHolder = {
                id: t('mobile.system_message.update_channel_header_message_and_forget.updated_from'),
                defaultMessage: '{username} updated the channel header from: {oldHeader} to: {newHeader}',
            };

            values = {username, oldHeader, newHeader};
            return renderMessage(postBodyProps, styles, intl, localeHolder, values);
        }

        localeHolder = {
            id: t('mobile.system_message.update_channel_header_message_and_forget.updated_to'),
            defaultMessage: '{username} updated the channel header to: {newHeader}',
        };

        values = {username, oldHeader, newHeader};
        return renderMessage(postBodyProps, styles, intl, localeHolder, values);
    } else if (postProps.old_header) {
        localeHolder = {
            id: t('mobile.system_message.update_channel_header_message_and_forget.removed'),
            defaultMessage: '{username} removed the channel header (was: {oldHeader})',
        };

        values = {username, oldHeader, newHeader};
        return renderMessage(postBodyProps, styles, intl, localeHolder, values);
    }

    return null;
};

const renderPurposeChangeMessage = (postBodyProps, styles, intl) => {
    const {postProps} = postBodyProps;
    let values;

    if (!postProps.username) {
        return null;
    }

    const username = renderUsername(postProps.username);
    const oldPurpose = postProps.old_purpose;
    const newPurpose = postProps.new_purpose;
    let localeHolder;

    if (postProps.new_purpose) {
        if (postProps.old_purpose) {
            localeHolder = {
                id: t('mobile.system_message.update_channel_purpose_message.updated_from'),
                defaultMessage: '{username} updated the channel purpose from: {oldPurpose} to: {newPurpose}',
            };

            values = {username, oldPurpose, newPurpose};
            return renderMessage(postBodyProps, styles, intl, localeHolder, values);
        }

        localeHolder = {
            id: t('mobile.system_message.update_channel_purpose_message.updated_to'),
            defaultMessage: '{username} updated the channel purpose to: {newPurpose}',
        };

        values = {username, oldPurpose, newPurpose};
        return renderMessage(postBodyProps, styles, intl, localeHolder, values);
    } else if (postProps.old_purpose) {
        localeHolder = {
            id: t('mobile.system_message.update_channel_purpose_message.removed'),
            defaultMessage: '{username} removed the channel purpose (was: {oldPurpose})',
        };

        values = {username, oldPurpose, newPurpose};
        return renderMessage(postBodyProps, styles, intl, localeHolder, values);
    }

    return null;
};

const renderDisplayNameChangeMessage = (postBodyProps, styles, intl) => {
    const {postProps} = postBodyProps;
    const oldDisplayName = postProps.old_displayname;
    const newDisplayName = postProps.new_displayname;

    if (!(postProps.username && postProps.old_displayname && postProps.new_displayname)) {
        return null;
    }

    const username = renderUsername(postProps.username);
    const localeHolder = {
        id: t('mobile.system_message.update_channel_displayname_message_and_forget.updated_from'),
        defaultMessage: '{username} updated the channel display name from: {oldDisplayName} to: {newDisplayName}',
    };

    const values = {username, oldDisplayName, newDisplayName};
    return renderMessage(postBodyProps, styles, intl, localeHolder, values);
};

const renderArchivedMessage = (postBodyProps, styles, intl) => {
    const {postProps} = postBodyProps;

    const username = renderUsername(postProps.username);
    const localeHolder = {
        id: t('mobile.system_message.channel_archived_message'),
        defaultMessage: '{username} archived the channel',
    };

    const values = {username};
    return renderMessage(postBodyProps, styles, intl, localeHolder, values);
};

const systemMessageRenderers = {
    [Posts.POST_TYPES.HEADER_CHANGE]: renderHeaderChangeMessage,
    [Posts.POST_TYPES.DISPLAYNAME_CHANGE]: renderDisplayNameChangeMessage,
    [Posts.POST_TYPES.PURPOSE_CHANGE]: renderPurposeChangeMessage,
    [Posts.POST_TYPES.CHANNEL_DELETED]: renderArchivedMessage,
};

export const renderSystemMessage = (postBodyProps, styles, intl) => {
    const renderer = systemMessageRenderers[postBodyProps.postType];
    if (!renderer) {
        return null;
    }
    return renderer(postBodyProps, styles, intl);
};
