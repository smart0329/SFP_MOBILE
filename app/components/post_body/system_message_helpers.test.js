// Copyright (c) 2019-present Smart.Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import * as SystemMessageHelpers from './system_message_helpers';
import {Posts} from 'mattermost-redux/constants';

const basePostBodyProps = {
    postProps: {
        username: 'username',
    },
    onPress: jest.fn(),
};

const mockStyles = {
    messageStyle: {},
    textStyles: {},
};

const mockIntl = {
    formatMessage: ({defaultMessage}) => defaultMessage,
};

describe('renderSystemMessage', () => {
    test('uses renderer for Channel Header update', () => {
        const postBodyProps = {
            ...basePostBodyProps,
            postProps: {
                ...basePostBodyProps.postProps,
                old_header: 'old header',
                new_header: 'new header',
            },
            postType: Posts.POST_TYPES.HEADER_CHANGE,
        };
        const renderedMessage = SystemMessageHelpers.renderSystemMessage(postBodyProps, mockStyles, mockIntl);
        expect(renderedMessage).toMatchSnapshot();
    });

    test('uses renderer for Channel Display Name update', () => {
        const postBodyProps = {
            ...basePostBodyProps,
            postProps: {
                ...basePostBodyProps.postProps,
                old_displayname: 'old displayname',
                new_displayname: 'new displayname',
            },
            postType: Posts.POST_TYPES.DISPLAYNAME_CHANGE,
        };

        const renderedMessage = SystemMessageHelpers.renderSystemMessage(postBodyProps, mockStyles, mockIntl);
        expect(renderedMessage).toMatchSnapshot();
    });

    test('uses renderer for Channel Purpose update', () => {
        const postBodyProps = {
            ...basePostBodyProps,
            postProps: {
                ...basePostBodyProps.postProps,
                old_purpose: 'old purpose',
                new_purpose: 'new purpose',
            },
            postType: Posts.POST_TYPES.PURPOSE_CHANGE,
        };

        const renderedMessage = SystemMessageHelpers.renderSystemMessage(postBodyProps, mockStyles, mockIntl);
        expect(renderedMessage).toMatchSnapshot();
    });

    test('uses renderer for archived channel', () => {
        const postBodyProps = {
            ...basePostBodyProps,
            postProps: {
                ...basePostBodyProps.postProps,
            },
            postType: Posts.POST_TYPES.CHANNEL_DELETED,
        };

        const renderedMessage = SystemMessageHelpers.renderSystemMessage(postBodyProps, mockStyles, mockIntl);
        expect(renderedMessage).toMatchSnapshot();
    });

    test('is null for non-qualifying system messages', () => {
        const postBodyProps = {
            ...basePostBodyProps,
            postType: 'not_relevant',
        };

        const renderedMessage = SystemMessageHelpers.renderSystemMessage(postBodyProps, mockStyles, mockIntl);
        expect(renderedMessage).toBeNull();
    });
});
