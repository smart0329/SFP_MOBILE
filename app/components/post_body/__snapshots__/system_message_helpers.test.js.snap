// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderSystemMessage uses renderer for Channel Display Name update 1`] = `
<Connect(Markdown)
  baseTextStyle={Object {}}
  onPostPress={[MockFunction]}
  textStyles={Object {}}
  value="{username} updated the channel display name from: {oldDisplayName} to: {newDisplayName}"
/>
`;

exports[`renderSystemMessage uses renderer for Channel Header update 1`] = `
<Connect(Markdown)
  baseTextStyle={Object {}}
  onPostPress={[MockFunction]}
  textStyles={Object {}}
  value="{username} updated the channel header from: {oldHeader} to: {newHeader}"
/>
`;

exports[`renderSystemMessage uses renderer for Channel Purpose update 1`] = `
<Connect(Markdown)
  baseTextStyle={Object {}}
  onPostPress={[MockFunction]}
  textStyles={Object {}}
  value="{username} updated the channel purpose from: {oldPurpose} to: {newPurpose}"
/>
`;

exports[`renderSystemMessage uses renderer for archived channel 1`] = `
<Connect(Markdown)
  baseTextStyle={Object {}}
  onPostPress={[MockFunction]}
  textStyles={Object {}}
  value="{username} archived the channel"
/>
`;
